NuRadioReco/modules/voltageToEfieldConverter.py

```diff
@@ -31,65 +32,60 @@ def get_array_of_channels(station, use_channels, det, zenith, azimuth,
         # determine refractive index of signal propagation speed between antennas
         refractive_index = ice.get_refractive_index(1, site)  # if signal comes from above, in-air propagation speed
         if station.is_cosmic_ray():
-            if(zenith > 0.5 * np.pi):
+            if zenith > 0.5 * np.pi:
                 refractive_index = ice.get_refractive_index(antenna_position[2], site)  # if signal comes from below, use refractivity at antenna position
+
         if station.is_neutrino():
             refractive_index = ice.get_refractive_index(antenna_position[2], site)
-        time_shift = -geo_utl.get_time_delay_from_direction(zenith, azimuth, antenna_position, n=refractive_index)
-        t_geos[iCh] = time_shift
-        t_cables[iCh] = channel.get_trace_start_time()
-        logger.debug("time shift channel {}: {:.2f}ns (signal prop), {:.2f}ns (trace start time)".format(channel.get_id(), time_shift, channel.get_trace_start_time()))
-        time_shift += channel.get_trace_start_time()
-        time_shifts[iCh] = time_shift
-
-    delta_t = time_shifts.max() - time_shifts.min()
-    tmin = time_shifts.min()
-    tmax = time_shifts.max()
-    logger.debug("adding relative station time = {:.0f}ns".format((t_cables.min() + t_geos.max()) / units.ns))
-    logger.debug("delta t is {:.2f}".format(delta_t / units.ns))
-    trace_length = station.get_channel(use_channels[0]).get_times()[-1] - station.get_channel(use_channels[0]).get_times()[0]
-    debug_cut = 0
-    if(debug_cut):
-        fig, ax = plt.subplots(len(use_channels), 1)
+
+        time_shift = -geo_utl.get_time_delay_from_direction(zenith, azimuth, antenna_position - efield_position, n=refractive_index)
+
+        t_shifts.append(time_shift)
+        t_min = channel.get_trace_start_time() + time_shift
+        t_mins.append(t_min)
+        t_max = t_min + channel.get_number_of_samples() / channel.get_sampling_rate()
+        t_maxs.append(t_max)
+
+    # take the intersection of all channels
+    t_min = np.max(t_min)
+    t_max = np.min(t_maxs)
+
+    n_samples = int((t_max - t_min) * channel.get_sampling_rate())
+    if n_samples % 2:
+        n_samples -= 1
+
+    window = NuRadioReco.framework.base_trace.BaseTrace()  # create base trace class to do the fft with correct normalization etc.
+    window.set_trace(np.zeros(n_samples), channel.get_sampling_rate(), t_min)
 
     traces = []
```

Comment:
In principle could create the appropriately sized array already as opposed to appending, but this is nitpicking.